### PR TITLE
Implement boolean operators

### PR DESCRIPTION
**Pull request by hellocoldworld**
Hello, added a new condition to the filter stage of my aggregation pipeline and suddendly my tests broke, because of the boolean operator.
I wanted to give it a try, but I'm afraid I'm being too simplistic maybe?
Or is this enough?

### CODE REVIEW

File: mongomock/aggregate.py [hunk 1]
```diff
@@ -271,6 +270,19 @@ def _parse_basic_expression(self, expression):
             return helpers.get_value_by_dot(self._doc_dict, expression[1:], can_generate_array=True)
         return expression
 
+    def _handle_boolean_operator(self, operator, values):
+        if operator == '$and':
+            return all([self.parse(value) for value in values])
```
Comment (pcorpet) [inline mongomock/aggregate.py]: Please use `_parse_to_bool`, same below.

File: tests/test__collection_api.py [hunk 2]
```diff
@@ -5611,6 +5607,56 @@ def test__aggregate_array_to_object(self):
             with self.assertRaises(mongomock.OperationFailure):
                 collection.aggregate(item)
 
+    def test_aggregate_project_with_boolean(self):
```
Comment (pcorpet) [inline tests/test__collection_api.py:5610]: Please also add tests in `test__mongomock` to make sure the behavior is consistent with pymongo + MongoDB

File: tests/test__collection_api.py [hunk 3]
```diff
@@ -5611,6 +5607,56 @@ def test__aggregate_array_to_object(self):
...
+        collection = self.db.collection
+        collection.insert_many([{
+            'items': [],
+        }])
+
+        expect = [{'items': True}]
+        actual = self.db.collection.aggregate([
+            {'$project': {'_id': 0, 'items': {'$and': [True, True]}}}
```
Comment (pcorpet) [inline tests/test__collection_api.py:5678]: Please add more tests:
- one test using more than 2 items
- one test using less than 2 items (even 0)
- one test using something else than booleans

You don't need to cover all combination though, it's OK to have point checks only.

**Review by pcorpet — CHANGES_REQUESTED** [review]
Thanks for adding this, it's exactly the right way.

I've added few comments in the tests, and I think you could also decide to cover extra cases where the syntax is not correct (for instance {'$and': 2}), but you can add a TODO and push those corner cases in a future PR.

Also make sure the CI passes, including the coverage check.
